io/disk/multipath_test.py

```diff
@@ -118,12 +120,64 @@ def test(self):
                 if not multipath.device_exists(path_dic["wwid"]):
                     msg += "Recovery of %s fails\n" % path_dic["wwid"]
 
-            # Blacklist sdX
-            for disk in path_dic["paths"]:
-                cmd = "devnode %s" % disk
-                multipath.form_conf_mpath_file(blacklist=cmd)
-                if disk in multipath.get_paths(path_dic["wwid"]):
-                    msg += "Blacklist of %s fails\n" % disk
+            # Failing and reinstating individual paths eg: sdX
+            self.log.info(" Failing and reinstating the individual paths")
+            for path in path_dic["paths"]:
+                if multipath.fail_path(path) is False:
+                    msg += "test failed while failing %s" % path
+                elif multipath.reinstate_path(path) is False:
+                    msg += "test failed while reinstating %s" % path
+        self.mpath_svc.restart()
+        # Failing n-1 paths for short time and reinstating back
+        self.log.info("Failing and reinstating the n-1 paths")
+        for wwid in self.wwids:
+            if wwid not in process.system_output('multipath -ll',
+                                                 ignore_status=True,
+                                                 shell=True):
+                continue
+            mpaths = multipath.get_paths(wwid)
+            count = 0
+            count = len(mpaths)-1
+            for path in range(0, count):
+                if multipath.fail_path(mpaths[path]) is False:
+                    msg += "%s did not failed in short fail" % path
+
+        time.sleep(330)
+        for wwid in self.wwids:
+            if wwid not in process.system_output('multipath -ll',
+                                                 ignore_status=True,
+                                                 shell=True):
+                continue
+            mpaths = multipath.get_paths(wwid)
```

Comment:
Do you need to get mpaths all the time ? I think it can be read once and stored, and used for subsequent tests. I am talking about:

```
        for wwid in self.wwids:
            if wwid not in process.system_output('multipath -ll',
                                                 ignore_status=True,
                                                 shell=True):
                continue
        mpaths = multipath.get_paths(wwid)
```

Reply:
yes this is required as we are trying to get the individual paths dynamically.
storing in some variable creates lots of mesh @narasimhan-v 
hope u got it.

Reply:
I still do not understand why we want to do it dynamically